sources/packages/backend/apps/api/src/route-controllers/application-offering-change-request/_tests_/e2e/application-offering-change-request.students.controller.updateApplicationOfferingChangeRequest.e2e-spec.ts

```diff
@@ -0,0 +1,242 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import {
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  FakeStudentUsersTypes,
+  getStudentToken,
+  getStudentByFakeStudentUserType,
+} from "../../../../testHelpers";
+import {
+  saveFakeApplicationOfferingRequestChange,
+  createE2EDataSources,
+  E2EDataSources,
+  saveFakeApplication,
+} from "@sims/test-utils";
+import {
+  ApplicationOfferingChangeRequestStatus,
+  ApplicationStatus,
+  Student,
+} from "@sims/sims-db";
+
+describe("ApplicationOfferingChangeRequestStudentsController(e2e)-updateApplicationOfferingChangeRequest", () => {
+  let app: INestApplication;
+  let student: Student;
+  let db: E2EDataSources;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+    student = await getStudentByFakeStudentUserType(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+      dataSource,
+    );
+  });
+
+  it("Should update the application offering change request status for the provided application offering change request id.", async () => {
+    // Arrange
+    const application = await saveFakeApplication(
+      db.dataSource,
+      {
+        student,
+      },
+      { applicationStatus: ApplicationStatus.Completed },
+    );
+    const applicationOfferingChangeRequest =
+      await saveFakeApplicationOfferingRequestChange(db, {
+        application,
+      });
+    const token = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+    const endpoint = `/students/application-offering-change-request/${applicationOfferingChangeRequest.id}`;
+    // Act/Assert
+    await request(app.getHttpServer())
+      .patch(endpoint)
+      .send({
+        studentConsent: true,
+        applicationOfferingChangeRequestStatus:
+          ApplicationOfferingChangeRequestStatus.InProgressWithSABC,
+      })
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK);
+  });
+
+  it("Should throw a HttpStatus Not Found (404) error when an application offering change update is requested for an application not in completed state.", async () => {
+    // Arrange
+    const application = await saveFakeApplication(
```

Comment:
just a suggestion, as all the application variable name is created as application, may cause confusion, it would be better to name something  as overwrittenApplication